frontend/__tests__/modals.test.js

```
@@ -0,0 +1,25 @@
+const fs = require('fs');
+const path = require('path');
+
+const html = fs.readFileSync(path.join(__dirname, '../index.html'), 'utf8');
+const originalJs = fs.readFileSync(path.join(__dirname, '../app.js'), 'utf8');
+const appJs = originalJs.replace(/const API_BASE.*\n/, "const API_BASE = '';\n");
+
+describe('Modal UI', () => {
+  beforeEach(() => {
+    document.documentElement.innerHTML = html.toString();
+    eval(appJs); // load patched script
```

Biome (1.9.4)

[error] eval() exposes to security risks and performance issues.

See the MDN web docs for more details.
Refactor the code so that it doesn't need to call eval().

(lint/security/noGlobalEval)